nf_core/components/components_utils.py

```diff
@@ -161,48 +162,35 @@ def get_components_to_install(
                 name, link = match.groups()
                 if link.startswith("../../../"):
                     name_split = name.lower().split("_")
-                    component_dict = {
-                        "name": "/".join(name_split),
-                        "org_path": None,
-                        "git_remote": None,
-                        "branch": None,
+                    component_name = "/".join(name_split)
+                    component_dict: Dict[str, Optional[str]] = {
+                        "name": component_name,
                     }
-                    modules.append(component_dict)
+                    modules[component_name] = component_dict
                 elif link.startswith("../"):
-                    component_dict = {"name": name.lower(), "org_path": None, "git_remote": None, "branch": None}
-                    subworkflows.append(component_dict)
+                    component_name = name.lower()
+                    component_dict = {"name": component_name}
+                    subworkflows[component_name] = component_dict
 
     if Path(subworkflow_dir, "meta.yml").exists():
         with open(Path(subworkflow_dir, "meta.yml")) as fh:
             meta = yaml.safe_load(fh)
-            if "components" not in meta:
-                return modules, subworkflows
-            components = meta.get("components")
-            new_module_list = []
-            new_subwf_list = []
-
-            for component in components:
-                if isinstance(component, str):
-                    component_dict = {"name": component, "org_path": None, "git_remote": None, "branch": None}
-                else:
-                    name = list(component.keys())[0]
-                    git_remote = component[name]["git_remote"]
-                    org_path_match = re.search(r"(?:https://|git@)[\w\.]+[:/](.*?)/", git_remote)
-                    org_path = org_path_match.group(1) if org_path_match else None
-                    component_dict = {
-                        "name": name.lower(),
-                        "org_path": org_path,
-                        "git_remote": git_remote,
-                        "branch": component[name].get("branch", "master"),
-                    }
-
-                if component_dict["name"] in [sw["name"] for sw in subworkflows]:
-                    new_subwf_list.append(component_dict)
-                else:
-                    if component_dict["name"] in [m["name"] for m in modules]:
-                        new_module_list.append(component_dict)
-
-        modules = new_module_list
-        subworkflows = new_subwf_list
-
-    return modules, subworkflows
+            if "components" in meta:
+                components = meta.get("components")
```

Comment:
You've checked just above that the key "components" is in the dictionary
```suggestion
                components = meta["components"]
```